lib/ruboty/github/actions/search_issues.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Ruboty
+  module Github
+    module Actions
+      class SearchIssues < Base
+        def call
+          if has_access_token?
+            search
+          else
+            require_access_token
+          end
+        end
+
+        private
+
+        def search
+          results = search_issues.items.map do |item|
+            [
+              "[#{item[:repository_url].delete_prefix('https://api.github.com/repos/')}##{item[:number]}] #{item[:title]} (#{item[:user][:login]})",
+              item[:html_url]
+            ].join("\n")
+          end
+          results.unshift("Searched: '#{query}'")
```

Comment:
> IMO & NITS:
> unshift method is not intuitive for the code flow, so I prefer the following code

This code is the same as for the other commands. (e.g. `create_deploy_pull_request`)

https://github.com/increments/ruboty-qiita-github/blob/20943ca1e3934884b7d9736530565058293f1a7f/lib/ruboty/github/actions/create_deploy_pull_request.rb#L20-L26

I agree that the unshift method is not intuitive for code flow. I will modify the code.

```suggestion
          results = search_issues.items.each_with_object(["Searched: '#{query}'"]) do |item, object|
            object << "[#{item[:repository_url].delete_prefix('https://api.github.com/repos/')}##{item[:number]}] #{item[:title]} (#{item[:user][:login]})\nitem[:html_url]"
          end
```

Reply:
> Could you write down an image of the message?

I envision the following message.
I refer to the message sent from the GitHub for Slack app (https://slack.github.com, https://github.com/integrations/slack#scheduled-reminders).

The reason for not keeping the format exactly the same is that ruboty is not used exclusively in Slack.

![image](https://user-images.githubusercontent.com/4934628/195471671-45627b03-091f-48c5-a7a2-8433d51e47d9.png)

```
Searched: 'org:increments is:open is:public is:pr'
[increments/ruboty-qiita-github#7] Add search issues command and refactoring (mziyut)
https://github.com/increments/ruboty-qiita-github/pull/7
[increments/qiita-markdown#122] Add Docswell to allowing embed script/iframe list (ku-suke)
https://github.com/increments/qiita-markdown/pull/122
[increments/chef-td-agent#4] Bump addressable from 2.6.0 to 2.8.1 (dependabot[bot])
https://github.com/increments/chef-td-agent/pull/4
[increments/qiitan-rb#22] Bump addressable from 2.5.2 to 2.8.1 (dependabot[bot])
https://github.com/increments/qiitan-rb/pull/22
```

Reply:
Changed the code to the following because of Rubocop's warning

```
Inspecting 16 files
..........C.....

Offenses:

lib/ruboty/github/actions/search_issues.rb:19:149: C: Layout/LineLength: Line is too long. [172/148]
            object << "[#{item[:repository_url].delete_prefix('https://api.github.com/repos/')}##{item[:number]}] #{item[:title]} (#{item[:user][:login]})\nitem[:html_url]"
                                                                                                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^

16 files inspected, 1 offense detected
```

https://github.com/increments/ruboty-qiita-github/blob/a3fb3554d9e7bcf5b4e8cfdb7070ae7f92120f88/lib/ruboty/github/actions/search_issues.rb#L15-L22